pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/logicaltable/BaseLogicalTableIntegrationTest.java

```diff
@@ -158,11 +171,70 @@ public void tearDown()
     cleanup();
   }
 
-  protected abstract List<String> getOfflineTableNames();
+  protected Map<String, List<File>> getOfflineTableDataFiles() {
+    List<String> offlineTableNames = getOfflineTableNames();
+    return !offlineTableNames.isEmpty() ? distributeFilesToTables(offlineTableNames, _avroFiles) : Map.of();
+  }
+
+  protected Map<String, List<File>> getRealtimeTableDataFiles() {
+    List<String> realtimeTableNames = getRealtimeTableNames();
+    return !realtimeTableNames.isEmpty() ? distributeFilesToTables(realtimeTableNames, _avroFiles) : Map.of();
+  }
+
+  protected Map<String, List<File>> distributeFilesToTables(List<String> tableNames, List<File> avroFiles) {
+    Map<String, List<File>> tableNameToFilesMap = new HashMap<>();
+
+    // Initialize the map with empty lists for each table name
+    tableNames.forEach(table -> tableNameToFilesMap.put(table, new ArrayList<>()));
+
+    // Round-robin distribution of files to table names
+    for (int i = 0; i < avroFiles.size(); i++) {
+      String tableName = tableNames.get(i % tableNames.size());
+      tableNameToFilesMap.get(tableName).add(avroFiles.get(i));
+    }
+    return tableNameToFilesMap;
+  }
+
+  protected List<String> getOfflineTableNames() {
```

Comment:
These functions should be abstract.

Reply:
I did not make them abstract to reduce the code duplication and each of the tests are overriding the necessary methods.

In general, all the super classes in the test framework have implementations of the methods and which can be overridden by further more specific test classes.

Let me know if you want me to refactor and make them abstract. I do understand the readability aspect but it should be easy when someone starts from the setup methods.